vm/contract/abi/abi_ptmkey.go

```diff
@@ -0,0 +1,127 @@
+package abi
+
+import (
+	_ "errors"
+	"strings"
+	"fmt"
+	_ "github.com/qlcchain/go-qlc/common/statedb"
+	"github.com/qlcchain/go-qlc/common/types"
+
+	"github.com/qlcchain/go-qlc/common/util"
+	"github.com/qlcchain/go-qlc/common"
+	"github.com/qlcchain/go-qlc/common/vmcontract/contractaddress"
+	_ "github.com/qlcchain/go-qlc/config"
+	"github.com/qlcchain/go-qlc/vm/abi"
+	"github.com/qlcchain/go-qlc/vm/vmstore"
+)
+
+const (
+	JsonPtmKey = `[
+		{"type":"function","name":"PtmKeyDeleteByBtype","inputs":[
+			{"name":"account","type":"string"},	
+			{"name":"btype","type":"uint32"}
+		]},
+		{"type":"function","name":"PtmKeyUpdate","inputs":[
+			{"name":"account","type":"string"},
+			{"name":"btype","type":"uint32"},
+			{"name":"pubkey","type":"string"}
+		]}
+	]`
+
+	MethodNamePtmKeyDeleteByBtype	 		= "PtmKeyDeleteByBtype"
+	MethodNamePtmKeyUpdate    				= "PermissionNodeUpdate"
+	VariableNamePtmKeyInfo      			= "PtmKeyInfo"
+)
+
+var (
+	PtmKeyABI, _ = abi.JSONToABIContract(strings.NewReader(JsonPtmKey))
+)
+
+const (
+	// contractAddr(32) + account(32) + btype(4) => pubkey + valid
+	PtmKeyAccIndexS  = types.AddressSize + 1
+	PtmKeyAccIndexE  = PtmKeyAccIndexS  + 32
+	PtmKeyBtypeIndexS = PtmKeyAccIndexE + 1
+	PtmKeyBtypeIndexE = PtmKeyBtypeIndexS + 4
+)
+
+type PtmKeyInfo struct {
+	Account types.Address
+	VBtype   uint32
+	VKey    []byte
+}
+
+type PtmKeyDeleteInfo struct {
+	Account types.Address
+	VBtype   uint32
+}
+
+type PtmKeyStorage struct {
+	Account types.Address
+	VBtype   uint32
+	VKey  []byte
+	Valid bool
+}
+
+func PtmKeyPledgeCheck(ctx *vmstore.VMContext, account types.Address) error {
+	_, err := ctx.Ledger.GetAccountMeta(account)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func PtmKeyInfoCheck(ctx *vmstore.VMContext, account types.Address, pt uint32,  pk []byte) error {
+	switch pt {
+		case common.PtmKeyVBtypeDefault:
+			if len(pk) != 44 {
+				return fmt.Errorf("pk len err")
+			}
+		default:
+			return fmt.Errorf("puslish type(%s) err", common.PtmKeyBtypeToString(pt))
+	}
+
+	return nil
+}
+
+func GetPtmKeyByAccountAndBtype(ctx *vmstore.VMContext, account types.Address, vBtype uint32) (*PtmKeyInfo, error) {
+	var key []byte
+	key = append(key, account[:]...)
+	key = append(key, util.BE_Uint32ToBytes(vBtype)...)
+	val, err := ctx.GetStorage(contractaddress.PtmKeyKVAddress[:], key)
+	if err != nil {
+		return nil, err
+	}
+
+	pkstorage := new(PtmKeyStorage)
+	err = PtmKeyABI.UnpackVariable(pkstorage, VariableNamePtmKeyInfo, val)
+	if err != nil || !pkstorage.Valid {
+		return nil, err
+	}
+	pk := new(PtmKeyInfo)
+	pk.Account = pkstorage.Account
+	pk.VBtype = pkstorage.VBtype
+	pk.VKey = pkstorage.VKey
+	return pk, nil
+}
+
+func GetPtmKeyByAccount(ctx *vmstore.VMContext, account types.Address) (*PtmKeyInfo, error) {
+	var key []byte
+	key = append(key, account[:]...)
+	
+	val, err := ctx.GetStorage(contractaddress.PtmKeyKVAddress[:], key)
```

Comment:
There are many keys for one account, key is account + type, use Iterator to return array, or use default type.